### ISSUE TITLE
Files are accidentally combined when ext option

### ISSUE THREAD

**Issue by helgatheviking** [gruntjs/grunt-contrib-uglify#341]
When I try to find use the bit of code from the readme to dynamically uglify all the files in a folder:
```
// Project configuration.
grunt.initConfig({
  uglify: {
    my_target: {
      files: [{
          expand: true,
          cwd: 'src/js',
          src: '**/*.js',
          ext: '.min.js',
          dest: 'dest/js'
      }]
    }
  }
});
```

If the filenames have `.` in them uglify gets confused and concatenates files. For example: I have `jquery.fitvids.js` and `jquery.fastclick.js` and after running the uglify from above I end up only with `jquery.min.js`. However, if I drop the `ext` setting then I get 3 files as expected.

**Comment by xPetrus** [comment]
I replicated your problem and has the same issue, it only worked correctly when I renamed the files without "jquery.". Just leave the files as .js extensions.

**Comment by helgatheviking** [comment]
Yes, it works when I rename the files to remove the "jquery". I think it would be ideal for the grunt plugin to get the file name from whatever preceeds `.js` instead of first character up until the first `.`

**Comment by skip405** [comment]
Can you confirm it combines the files? Not just overwrites them with the last one it uglified. E.g. uglifying `jquery.plugin1.js` and `jquery.plugin2.js` will result in code from *2* files being present in the resulting file `jquery.min.js`, not just from `plugin2`

**Comment by XhmikosR** [comment]
You are welcome to make a PR.
